addons/project/models/project.py

```diff
@@ -766,6 +766,7 @@ def create(self, vals):
             vals.update(self.update_date_end(vals['stage_id']))
             vals['date_last_stage_update'] = fields.Datetime.now()
         task = super(Task, self.with_context(context)).create(vals)
+        task._portal_ensure_token()
```

Comment:
Why so ? It is already called in various methods, notably _get_share_url. Where is it incorrectly computed ?

Reply:
I will add to this question the general rule: an access token should never be generated until it is necessary, because generating it is already opening a door.

If it is absolutely necessary to generate the token at creation on a specific flux, it should be done after the create on that specific flux.

Usually however the token only becomes necessary when the record "share url" is about to be sent externally (often by email).

Reply:
Totally agree with @seb-odoo :) (+ db storage space)

Reply:
Indeed, and we do not need this anymore if we don't always want a token every time, because the first fix fix also fix this model